newton/_src/geometry/contact_match.py

```
@@ -0,0 +1,496 @@
+        new_pos_w = wp.transform_point(data.body_q[bid], data.new_point0[tid])
+    new_n = data.new_normal[tid]
+
+    # Binary search for the first old contact with matching key.
+    start = _binary_search_int64(0, n_old, target_key, data.prev_keys)
+
+    if start == -1:
+        data.match_index[tid] = MATCH_NOT_FOUND
+        return
+
+    # Linear scan through contacts sharing the same key.
+    best_idx = int(-1)
```

Ruff (0.15.10)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)